controlplane/api/v1beta1/kthreescontrolplane_types.go

```diff
@@ -67,6 +67,17 @@ type KThreesControlPlaneSpec struct {
 	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
 	// +optional
 	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
+
+	// MachineTemplate contains information about how machines
+	// should be shaped when creating or updating a control plane.
+	MachineTemplate KThreesControlPlaneMachineTemplate `json:"machineTemplate,omitempty"`
+}
+
+type KThreesControlPlaneMachineTemplate struct {
```

Comment:
Should we leave a comment that mentions in the next api version we will move the InfrastructureTemplate field into this struct?